spec/features/search_page_spec.rb

```diff
@@ -4,113 +4,71 @@
-
-    dataset = DatasetBuilder.new
-                .with_title(dataset_title)
-                .build
-
+    dataset = build :dataset, title: 'a title'
     index(dataset)
-    search_for(query)
+    search_for('title')
 
     expect(page).to have_css('h1', text: 'Search results')
-    expect(page).to have_css('a', text: dataset_title)
+    expect(page).to have_css('a', text: 'title')
   end
 
   scenario 'Search results are correctly sorted' do
-    old_dataset = DatasetBuilder.new
-                    .with_title('Old Interesting Dataset')
-                    .with_name('old-dataset')
-                    .public_updated_at(Time.now - 1)
-                    .build
-
-    new_dataset = DatasetBuilder.new
-                    .with_title('Recent Interesting Dataset')
-                    .with_name('new-dataset')
-                    .build
-
+    old_dataset = build :dataset, title: 'Old Dataset', public_updated_at: 1.hour.ago
+    new_dataset = build :dataset, title: 'New Dataset'
     index(old_dataset, new_dataset)
 
-    search_for('Old Interesting Dataset')
-
+    search_for('Old dataset')
     expect(page).to have_css('option[selected]', text: 'Best match')
 
     elements = all('h2 a')
-
     expect(elements[0]).to have_content 'Old'
-    expect(elements[1]).to have_content 'Recent'
-
-    filtered_search_for('Interesting Dataset', 'Most recent')
+    expect(elements[1]).to have_content 'New'
 
+    filtered_search_for('Dataset', 'Most recent')
     expect(page).to have_css('option[selected]', text: 'Most recent')
 
     elements = all('h2 a')
-
-    expect(elements[0]).to have_content 'Recent'
+    expect(elements[0]).to have_content 'New'
     expect(elements[1]).to have_content 'Old'
 
-    filtered_search_for('Old Interesting Dataset', 'Best match')
-
+    filtered_search_for('Old Dataset', 'Best match')
     expect(page).to have_css('option[selected]', text: 'Best match')
 
     elements = all('h2 a')
     expect(elements[0]).to have_content 'Old'
-    expect(elements[1]).to have_content 'Recent'
+    expect(elements[1]).to have_content 'New'
   end
 
   scenario 'Match publisher query against available publishers', js: true do
-    first_dataset = DatasetBuilder.new
-                      .with_title('Data About Tonka Trucks')
-                      .with_publisher('Tonka Trucks')
-                      .build
-
-    second_dataset = DatasetBuilder.new
-                      .with_title('Data About Toby')
-                      .with_publisher('Toby Corp')
-                      .build
-
-    index(first_dataset, second_dataset)
+    dataset1 = build :dataset, title: 'Wanted'
+    dataset2 = build :dataset, :unrelated, title: 'Unrelated'
 
+    index(dataset1, dataset2)
     visit('/search')
-
     assert_data_set_length_is(2)
 
     execute_script "window.location = '#publisher'"
-
-    find_field('publisher').send_keys('Tob', :down, :enter)
+    find_field('publisher').send_keys(dataset1.organisation.title, :down, :enter)
```

Comment:
This test is simulating "Tob" being typed, and the autocomplete behaviour matching the full name of the publisher "Toby Corp". We've not lost that intent.

Reply:
Good point - done.